src/store/storeConfig.ts

```diff
@@ -243,6 +243,8 @@ export const storeConfig: StoreOptions<State> = {
         const newStoredMesh = ClientStorage.get('selectedMesh')
         if ((newStoredMesh === null || newStoredMesh === 'all') && state.meshes.items.length > 0) {
           dispatch('updateSelectedMesh', state.meshes.items[0].name)
+        } else if (newStoredMesh !== null && !state.meshes.items.some((mesh) => mesh.name === newStoredMesh)) {
+          dispatch('updateSelectedMesh', null)
```

Comment:
Kinda not related to this PR, but how does 'all' come into this? Is that like a backcompat/upgrade thing? Unlikely, but if I have a mesh called `all` it gets reset to the first mesh in the list. Not a blocker as it's unlikely, but just to explain as to why the `all` value got me wondering.

Reply:
Yeah, this is purely an artifact of removing a feature. You used to be able to select "all" in the mesh selector which would remove any views being mesh specific. E.g. the data plane list view would show _all_ data plane proxies for _all_ meshes.

That being said, I think the newly added logic would also handle this, so maybe I can replace the special casing of all with it.

Reply:
Gotcha, thanks for the explanation

Reply:
Simplified this in https://github.com/kumahq/kuma-gui/pull/493/commits/6e640aecd24cd79fba48c56508e2b7237ea35258.

Reply:
Ah wait looking at the previous commit 😆 all good! ... I think 👀

Reply:
Yeah defo think a test, I think we can get to an undefined `state.meshes.items` here? A test would definitely help I think.

Reply:
How can `state.meshes.items` become `undefined`? Even if `items` is omitted in the response, `fetchMeshList` has logic against this ensuring that `items` is always an array.

One thing that can technically happen is the list being empty and the fallback logic trying to pick its first element. I added tests for this and updated the logic.

Reply:
Yeah that's what I meant 😸 

> looks like state.meshes.items can be zero length?

Thanks for adding that check and the tests!